### ISSUE TITLE
Interop with `nodemon --exec` and `ncc run`

### ISSUE THREAD

**Issue by adamelliotfields** [zeit/ncc#253]
Nodemon sends the `SIGUSR2` signal when a watched file changes, which can be handled in your app to gracefully shutdown.

Since NCC does not listen for `SIGUSR2`, it will error:

```
Error: Application at src/index.js is already running or didn't cleanup after previous run.To manually clear the last runbuild, try running "rm -rf /tmp/d41d8cd98f00b204e9800998ecf8427e".
[nodemon] app crashed - waiting for file changes before starting...
```

In #228, listeners for `SIGINT` and `SIGTERM` were added. I'd like to add a listener for `SIGUSR2` as well (I've tested this locally and it resolves the issue).

Note that `ncc run --watch` won't work for me because [`ps.kill()`](https://github.com/zeit/ncc/blob/0.13.0/src/cli.js) will send a `SIGTERM` signal to my app, which will cause the process to exit, which stops NCC.

### Minimal Reproducible Example

Copy the following files, run `npm install`, and then `npm run start:ncc`. While the app is running, make a change to `src/index.js` and observe the results.

Run `npm run start:nodemon` and make a change to observe the desired result.

**package.json**

```json
{
  "private": true,
  "scripts": {
    "start:ncc": "nodemon --exec \"ncc run\" src/index.js",
    "start:nodemon": "nodemon src/index.js"
  },
  "dependencies": {
    "express": "^4"
  },
  "devDependencies": {
    "@zeit/ncc": "^0",
    "nodemon": "^1"
  }
}
```

**src/index.js**

```javascript
const express = require('express');
const http = require('http');

const app = express();

app.get('/', (req, res) => res.status(200).json({
  message: http.STATUS_CODES[200],
  statusCode: 200,
}));

const server = http.createServer(app);

['SIGINT', 'SIGTERM', 'SIGUSR2'].forEach((signal) => {
  process.on(signal, () => {
    console.log('Shutting down...');
    server.close(() => {
      process.exit(0);
    });
  });
});

server.listen({ host: 'localhost', port: 8080 }, () => {
  console.log('Listening on http://localhost:8080...');
});
```

**Comment by adamelliotfields** [comment]
Thanks for the quick reply, @guybedford!

Adding `process.on("SIGUSR2", exit);` to `cli.js` will be a quick fix.

I'll play around with `cli.js` locally and see if I can get `ncc run` to not exit when the child process exits. I think it's the [`ps.on("exit")`](https://github.com/zeit/ncc/blob/0.13.0/src/cli.js#L246) call.

Per the [docs](https://nodejs.org/api/child_process.html#child_process_event_exit):

> The 'exit' event is emitted after the child process ends. If the process exited, code is the final exit code of the process, otherwise null. If the process terminated due to receipt of a signal, signal is the string name of the signal, otherwise null. One of the two will always be non-null.

So maybe if we detect that the signal was one of `SIGINT`, `SIGTERM`, or `SIGUSR2`, we don't call `ps.on("exit")`?

**Comment by guybedford** [comment]
Thanks for the suggestions @adamelliotfields.

I started looking into the `run --watch` case, but get the impression there will be other edge cases that would also need to be managed here. For this reason I've posted https://github.com/zeit/ncc/pull/259 to throw on this case for now as unsupported, but we can certainly come back around on it as a feature.

**Comment by adamelliotfields** [comment]
Sounds good @guybedford. For now, Nodemon works for me. `--watch` would be a nice to have, but not a show-stopper.
